File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int temp = nums[0], cnt = 1;
        for(int i=1;i<nums.size();++i){
            if(nums[i]==temp){
                ++cnt;
            }else{
                --cnt;
                if(cnt==0){
                    temp=nums[i];
                    cnt=1;
                }
            }
            // cout<<cnt<<nums[i]<<endl;
        }
        return temp;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        result, count = nums[0], 1
        for i in range(1, len(nums)):
            if nums[i] == result:
                count += 1
            else:
                count -= 1
            if count == 0:
                result = nums[i+1]
        return result